### PR TITLE
fix: disable delete buttons in case deployment is active or session is running

### PR DESCRIPTION
**Pull request by RonenMars**
## What type of PR is this? (check all applicable)

- [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
- [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
- [x] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
- [ ] 🏎 (perf) - Optimization
- [ ] 📄 (docs) - Documentation - Documentation only changes
- [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
- [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ]  ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
- [ ]  ☑️ (chore) - Chores - Other changes that don't modify src or test files
- [ ]  ↩️ (revert) - Reverts - Reverts a previous commit(s).

## Description
Disable delete buttons for deployments/sessions if the deployment is active or the session running

### CODE REVIEW

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx [hunk 1]
```diff
@@ -116,6 +116,16 @@ export const AKDeploymentTableBody = ({ deployments }: { deployments?: Deploymen
 		hidePopper();
 	};
 
+	const deleteDeploymentDisplayPopper = (event: React.MouseEvent<HTMLDivElement>, deployment: Deployment) => {
+		if (deployment.state === DeploymentState.ACTIVE_DEPLOYMENT) {
```
Comment (efiShtain) [inline vscode-react/src/components/aKDeploymentTableBody.component.tsx:120]: so I just click and nothing happens ?

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx [hunk 2]
```diff
@@ -188,13 +198,10 @@ export const AKDeploymentTableBody = ({ deployments }: { deployments?: Deploymen
...
-						onClick={(e) => {
-							const refElement = e.currentTarget;
-							showPopper("deploymentDelete");
-							deletePopperElementRef.current = refElement;
-							setDeleteDeploymentId(deployment.deploymentId);
-						}}
+						className={`relative codicon codicon-trash ${
+							deployment.state === DeploymentState.ACTIVE_DEPLOYMENT ? "cursor-not-allowed" : "cursor-pointer"
+						} ml-2 z-20`}
+						onClick={(event) => deleteDeploymentDisplayPopper(event, deployment)}
```
Comment (efiShtain) [inline vscode-react/src/components/aKDeploymentTableBody.component.tsx:204]: you mean show delete deployment popper ? 
or you mean you delete the popper ?

File: vscode-react/src/components/aKSessionTableBody.component.tsx [hunk 3]
```diff
@@ -70,6 +71,16 @@ export const AKSessionsTableBody = ({
 		hidePopper();
 	};
 
+	const displaySessionDeletePopper = (event: React.MouseEvent<HTMLDivElement>, session: Session) => {
+		if (session.state === SessionState.RUNNING) {
```
Comment (efiShtain) [inline vscode-react/src/components/aKSessionTableBody.component.tsx:75]: again, nothing happens ? no user feedback ? 
why do you even allow the button in this case ?

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx [hunk 4]
```diff
@@ -116,6 +116,16 @@ export const AKDeploymentTableBody = ({ deployments }: { deployments?: Deploymen
...
 	};
 
+	const deleteDeploymentDisplayPopper = (event: React.MouseEvent<HTMLDivElement>, deployment: Deployment) => {
+		if (deployment.state === DeploymentState.ACTIVE_DEPLOYMENT) {
+			return;
+		}
+		const refElement = event.currentTarget;
+		showPopper("deploymentDelete");
+		deletePopperElementRef.current = refElement;
+		setDeleteDeploymentId(deployment.deploymentId);
```
Comment (efiShtain) [inline vscode-react/src/components/aKDeploymentTableBody.component.tsx:126]: what setDeleteDeploymentId do ?

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx (hunk 1 above)
Comment (RonenMars) [reply vscode-react/src/components/aKDeploymentTableBody.component.tsx:120]: Yes, there is an indication with the mouse pointer that this action is not allowed, take a look here: https://tailwindcss.com/docs/cursor for `cursor-not-allowed`

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx (hunk 2 above)
Comment (RonenMars) [reply vscode-react/src/components/aKDeploymentTableBody.component.tsx]: renamed

File: vscode-react/src/components/aKSessionTableBody.component.tsx (hunk 3 above)
Comment (RonenMars) [reply vscode-react/src/components/aKSessionTableBody.component.tsx:75]: There is user feedback as described here: https://github.com/autokitteh/vscode-extension/pull/214#discussion_r1549561541

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx (hunk 4 above)
Comment (RonenMars) [reply vscode-react/src/components/aKDeploymentTableBody.component.tsx:126]: The flow goes like this:
1. The user clicks the delete icon, which opens the deletePopper.
2. When the popper is opened, the `deploymentId` is being saved in the local state for further action.
3. When the user approves that he wants to delete the deployment, the request for the extension is being sent with the `deploymentId` saved when the modal has been opened: https://github.com/autokitteh/vscode-extension/blob/bc315de55bc59708a99a7daccd0bf30a2bdd407e/vscode-react/src/components/aKDeploymentTableBody.component.tsx#L109C1-L110C1

So we are saving the deploymentId to send it when the user approves the deletion.

File: vscode-react/src/components/aKDeploymentTableBody.component.tsx (hunk 1 above)
Comment (RonenMars) [reply vscode-react/src/components/aKDeploymentTableBody.component.tsx:122]: Done

File: vscode-react/src/components/aKSessionTableBody.component.tsx (hunk 3 above)
Comment (RonenMars) [reply vscode-react/src/components/aKSessionTableBody.component.tsx:75]: refactored as described above

**Review by efiShtain — APPROVED** [review]

**Pull request merged**